vars/pipelineApprobation.groovy

```diff
@@ -0,0 +1,132 @@
+                    //
+                    // GIT clone
+                    //
+                    stage('Git Clone') {
+                        parallel([
+                            'vega core': {
+                                dir('vega') {
+                                    gitClone('vega', params.VEGA_CORE_BRANCH)
+                                }
+                            },
+                            'specs-internal': {
+                                dir('specs-internal') {
+                                    gitClone('specs-internal', params.SPECS_INTERNAL_BRANCH)
+                                }
+                            },
+                            'MultisigControl': {
+                                dir('MultisigControl') {
+                                    gitClone('MultisigControl', params.MULTISIG_CONTROL_BRANCH)
+                                }
+                            },
+                            'system-tests': {
+                                dir('system-tests') {
+                                    gitClone('system-tests', params.SYSTEM_TESTS_BRANCH)
+                                }
+                            }
+                        ])
+                    }
+                    //
+                    // RUN
+                    //
+                    stage('Run Approbation') {
+                        sh label: 'approbation', script: """#!/bin/bash -e
+                            npx @vegaprotocol/approbation@${params.APPROBATION_VERSION} check-references \
+                                --specs="${params.SPECS_ARG}" \
+                                --tests="${params.TESTS_ARG}" \
+                                --ignore="${params.IGNORE_ARG}" \
+                                ${params.OTHER_ARG}
+                        """
+                    }
+                }
+                // Workaround Jenkins problem: https://issues.jenkins.io/browse/JENKINS-47403
+                // i.e. `currentResult` is not set properly in the finally block
+                // CloudBees workaround: https://support.cloudbees.com/hc/en-us/articles/218554077-how-to-set-current-build-result-in-pipeline
+                currentBuild.result = currentBuild.result ?: 'SUCCESS'
+                // result can be SUCCESS or UNSTABLE
+            } catch (FlowInterruptedException e) {
+                currentBuild.result = 'ABORTED'
+                throw e
+            } catch (e) {
+                // Workaround Jenkins problem: https://issues.jenkins.io/browse/JENKINS-47403
+                // i.e. `currentResult` is not set properly in the finally block
+                // CloudBees workaround: https://support.cloudbees.com/hc/en-us/articles/218554077-how-to-set-current-build-result-in-pipeline
+                currentBuild.result = 'FAILURE'
+                throw e
+            } finally {
+                stage('Cleanup') {
+                    echo 'Post something to Slack?'
```

Comment:
This works for now. The main output is way too long to post, so this works nicely as an indicator of where we *might* post something. I need to format some output more appropriately before we pick this up, so it doesn't need to be addressed in the initial merge

Reply:
we can do this in separate PR, it is also simple: call a function with a channel name and a message.